src/host/random.jl

```diff
@@ -97,8 +97,9 @@ function Random.randn!(rng::RNG, A::AnyGPUArray{T}) where T <: Number
     @kernel function randn!(a, randstates)
         i = @index(Global, Linear)
         idx = 2*(i - 1) + 1
-        U1 = gpu_rand(T, i, randstates)
-        U2 = gpu_rand(T, i, randstates)
+        threadidx = (i - 1) % length(randstates) + 1
```

Comment:
Why the clamping? It should be possible to use the thread index directly, as we did before: https://github.com/JuliaGPU/GPUArrays.jl/pull/559/files#diff-8c6d93e776a290aa5986cc1760bf0161b72c15994e11aefd7a2fd4b05497a5d8R34